pkg/sources/generator/tickgen.go

```diff
@@ -132,6 +135,21 @@ func NewMemGen(ctx context.Context, vertexInstance *dfv1.VertexInstance, opts ..
 		jitter = vertexInstance.Vertex.Spec.Source.Generator.Jitter.Duration
 	}
 
+	var genFunction = recordGenerator
+	if vertexInstance.Vertex.Spec.Source.Generator.ValueBlob != nil {
+		logger.Info("ValueBlob was set, using provided value instead of randomly generated data")
+		emitBytes, err := base64.StdEncoding.DecodeString(*vertexInstance.Vertex.Spec.Source.Generator.ValueBlob)
+		if err != nil {
+			logger.Errorw("Error decoding provided ValueBlob as base64, cannot initialize generator", zap.Error(err))
```

Comment:
In this case, shall we return the error?

Reply:
Indeed, fixed to include return

Reply:
done